### ISSUE TITLE
Loading of quantized dnn not possible

### ISSUE THREAD

**Issue by gorbad-6** [opencv/opencv#10235]
Currently, it is not possible to load quantized networks into opencv.

##### System information (version)

- OpenCV => 3.3.1-dev
- Operating System / Platform => Ubuntu 16.04
- Compiler => gcc 5+

##### Detailed description

After quantization network does not load into opencv.

##### Steps to reproduce

``` python
import os
import shutil
import numpy as np
import tensorflow as tf
import network.unet
import network.losses
import keras.backend as K
from tensorflow.python.framework import graph_io
from keras.layers import Input, BatchNormalization, Concatenate, Conv2D
from keras.models import Model
class MyUnet:
    @staticmethod
    def conv(x, nf, sz=3, pad='same',  act='relu', kern='he_normal', stride=1):
            return Conv2D(nf, sz, padding=pad, activation=act, kernel_initializer=kern, strides=(stride, stride))(x)

    @staticmethod
    def down_pass(x, nf):
            x = MyUnet.conv(x, nf)
            x = MyUnet.conv(x, nf)
            return x

    @staticmethod
    def toy_unet(in_shape = (32, 32, 3)):
            inputs = Input(in_shape)
            conv1 = MyUnet.down_pass(BatchNormalization()(inputs, training=False), 8)
            conv2 = MyUnet.down_pass(conv1, 16)
            conv3 = Conv2D( 3, 1, activation='sigmoid')(conv2)
            return Model( inputs, conv3 )

sess = tf.Session()
K.set_session(sess)
my_net = MyUnet.toy_unet()
my_net.compile(optimizer='rmsprop', loss='binary_crossentropy', metrics=['accuracy'])
data = np.random.random((10, 32, 32, 3))
labels=np.random.randint(3, size=(10, 32, 32, 3))
my_net.fit(data, labels, epochs=1, batch_size=5)
temp_export_dir = '/temp_export/'
if os.path.exists(temp_export_dir):
    shutil.rmtree(temp_export_dir)
os.makedirs(temp_export_dir)
checkpoint_prefix = os.path.join(temp_export_dir, "saved_checkpoint")
input_graph_name ='graph.pb'
with sess.graph.as_default():
    saver = tf.train.Saver()
    cp_path = saver.save(sess, checkpoint_prefix)
    graph_io.write_graph(sess.graph_def, temp_export_dir, input_graph_name)
```

``` bash
/tensorflow/bazel-bin/tensorflow/python/tools/freeze_graph --input_graph=/temp_export/graph.pb --input_checkpoint=/temp_export/saved_checkpoint --output_node_names=conv2d_5/Sigmoid --output_graph=/temp_export/frozen_graph.pb
```

This transformation works fine
```bash
/tensorflow/bazel-bin/tensorflow/tools/graph_transforms/transform_graph --in_graph=/temp_export/frozen_graph.pb --out_graph=/temp_export/frozen_opt.pb --inputs=input_1 --outputs=conv2d_5/Sigmoid --transforms="fold_batch_norms fold_old_batch_norms fold_constants strip_unused_nodes sort_by_execution_order"
```
i.e. model loads and forward pass is possible.

However, when I try to quantize the weights:
```bash
/tensorflow/bazel-bin/tensorflow/tools/graph_transforms/transform_graph --in_graph=/temp_export/frozen_graph.pb --out_graph=/temp_export/frozen_opt.pb --inputs=input_1 --outputs=conv2d_5/Sigmoid --transforms="fold_batch_norms fold_old_batch_norms fold_constants strip_unused_nodes quantize_weights sort_by_execution_order"
```
I get following error when trying to load the model

> conv2d_3/kernel/read/_5__cf__5:Dequantize(conv2d_3/kernel/read/_5__cf__5_quantized_const)(conv2d_3/kernel/read/_5__cf__5_quantized_min)(conv2d_3/kernel/read/_5__cf__5_quantized_max)
T:0
mode:[ ]
OpenCV Error: Unspecified error (Unknown layer type Dequantize in op conv2d_3/kernel/read/_5__cf__5) in populateNet, file /home/zsolt/Downloads/dockerfile/opencv_complete/opencv/modules/dnn/src/tensorflow/tf_importer.cpp, line 1299
<

**Comment by dkurt** [comment]
@gorbad-6, Could you test the model with https://github.com/opencv/opencv/pull/10402?

**Comment by gorbad-6** [comment]
@dkurt yes, thank you runs and predicts correctly. awesome!

**Comment by dkurt** [comment]
@gorbad-6, Could you share please some numbers of how much compression rate you usually have versus an accuracy loss during quantization? Thanks for your time!

**Comment by gorbad-6** [comment]
@dkurt I am not sure I got your question correctly.
I use the tensorflow transform-graph tool for quantization. The compression that I get from this is 
138Mb -> 34Mb
The accuracy loss is around 2e-5 per pixel (I do semantic segmentation on 512x512 8UC1 images, so the total errors are of order O(1), completely acceptable ).

Runtimes for forward passes, however are approximately the same for quantizes/non-quantized nets.
(my net has around 36M floats/ints and a pass takes both nets around 5000 to 6000ms as measured by the cv::TickMeter )
Hope this helps
